handlers/upload.go

```diff
@@ -73,6 +73,35 @@ func HasContentType(r *http.Request, mimetype string) bool {
 	return false
 }
 
+func (r UploadVODRequest) IsProfileValid() bool {
+	// an empty profile is valid and tells us to use the default ABR ladder
+	if len(r.Profiles) == 0 {
+		return true
+	}
+
+	// a special case where only the bitrate is set which tells us to
+	// generate a profile that matches the input video's specs with the
+	// user specified target bitrate
+	if len(r.Profiles) == 1 {
+		profile := r.Profiles[0]
+		return profile.Bitrate > 0
+	}
+
+	// verify that width/height/bitrate is provided in cases where the
+	// user wants to use their own transcode profile
+	for _, profile := range r.Profiles {
+		if profile.Width == 0 && profile.Height != 0 {
+			return false
+		} else if profile.Width != 0 && profile.Height == 0 {
+			return false
+		} else if profile.Width != 0 && profile.Height != 0 && profile.Bitrate == 0 {
+			return false
+		}
```

Comment:
Could this be simplified to `profile.Width == 0 || profile.Height == 0 || profile.Bitrate == 0`?

Reply:
Yeah, I had a few other checks there that got removed. Simplified it further.